Library/Dependencies/napi/source/js_native_api_JavaScriptCore.cc

```diff
@@ -1,353 +1,912 @@
+JSObjectRef JSCCallAsConstructor(JSContextRef ctx, JSObjectRef constructor, size_t argumentCount, const JSValueRef arguments[], JSValueRef* exception) {
+
+  auto iter = constructorCB.find(constructor);
+  assert(iter != constructorCB.end());
+  ClassTable *cbInfo = iter->second;
+  
+  
+  assert(JSObjectIsConstructor(ctx, constructor));
+  OpaqueJSValue* jsValue = const_cast<OpaqueJSValue*>(JSObjectMake(ctx, cbInfo->classRef, nullptr));
+  napi_value value = reinterpret_cast<napi_value>(jsValue);
+  
+  FunctionTables[jsValue] = &cbInfo->FunctionTable;
+  
+  // callback will wrap the pointer itself
+  CallbackInfo callbackInfo;
+  callbackInfo.argc = argumentCount;
+  auto jsValueRefs = const_cast<JSValueRef*>(arguments);
+  auto jsValues = const_cast<OpaqueJSValue**>(jsValueRefs);
+  callbackInfo.argv = reinterpret_cast<napi_value*>(jsValues);
+  callbackInfo.thisArg = value;
+  callbackInfo.data = cbInfo->data;
+  callbackInfo.isConstructCall = true;
+  cbInfo->cb(cbInfo->env, reinterpret_cast<napi_callback_info>(&callbackInfo));
+  
+  return jsValue;
 }
 
-napi_status napi_get_arraybuffer_info(napi_env env,
-                                      napi_value arraybuffer,
-                                      void** data,
-                                      size_t* byte_length) {
-  return napi_ok;
+bool JSCSetProperty(JSContextRef ctx, JSObjectRef object, JSStringRef propertyName, JSValueRef value, JSValueRef* exception) {
+  assert(0);
+  return false;
 }
 
+JSValueRef JSCGetProperty(JSContextRef ctx, JSObjectRef object, JSStringRef propertyName, JSValueRef* exception) {
+  auto iter = FunctionTables.find(object);
+  assert(iter != FunctionTables.end());
+  
+  
+  char propertyStr[1024];
+  /*size_t length =*/JSStringGetUTF8CString(propertyName, propertyStr, sizeof(propertyStr));
+  
+  auto propertyIter = iter->second->properties.find(std::string(propertyStr));
+  assert(propertyIter != iter->second->properties.end());
+  
+  auto prop = propertyIter->second;
+  
+  CallbackInfo callbackInfo;
+  callbackInfo.newTarget = 0;
+  callbackInfo.argc = 0;
+  callbackInfo.argv = 0;
+  callbackInfo.thisArg = reinterpret_cast<napi_value>(object);
+  callbackInfo.data = prop.data;
+  callbackInfo.isConstructCall = false;
+  napi_value retValue = prop.getter(iter->second->env, reinterpret_cast<napi_callback_info>(&callbackInfo));
+  
+  return reinterpret_cast<JSValueRef>(retValue);
+}
 
-napi_status napi_get_cb_info(
-    napi_env env,               // [in] NAPI environment handle
-    napi_callback_info cbinfo,  // [in] Opaque callback-info handle
-    size_t* argc,      // [in-out] Specifies the size of the provided argv array
-                       // and receives the actual count of args.
-    napi_value* argv,  // [out] Array of values
-    napi_value* this_arg,  // [out] Receives the JS 'this' arg for the call
-    void** data) {         // [out] Receives the data pointer for the callback.
+napi_status napi_define_class(napi_env env,
```

Comment:
should probably put a comment that this function should be fixed

Reply:
I prefer to update the issue on GH.